test/e2e/e2e_test.go

```diff
@@ -124,4 +132,78 @@ var _ = Describe("LWS Operator", Ordered, func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("should expose metrics endpoint with TLS", func() {
+		var (
+			err                error
+			podName            = "curl-metrics-test"
+			containerName      = "curl-metrics"
+			certMountPath      = "/etc/lws/metrics/certs"
+			metricsServiceName = "lws-controller-manager-metrics-service"
+		)
+		By("Creating curl test pod")
+		ctx := context.TODO()
+		curlPod := testutils.MakeCurlMetricsPod(operatorNamespace)
+		defer clients.KubeClient.CoreV1().Pods(operatorNamespace).Delete(context.TODO(), "curl-metrics-test", metav1.DeleteOptions{})
+		_, err = clients.KubeClient.CoreV1().Pods(operatorNamespace).Create(ctx, &curlPod.Pod, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred(), "failed to create curl metrics test pod")
+
+		Eventually(func() error {
+			pod, err := clients.KubeClient.CoreV1().Pods(operatorNamespace).Get(ctx, podName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get pod: %w", err)
+			}
+			if pod.Status.Phase != corev1.PodRunning {
+				return fmt.Errorf("pod %q not ready, phase: %s", podName, pod.Status.Phase)
+			}
+			return nil
+		}, operatorReadyTime, operatorPoll).Should(Succeed(), "curl-metrics-test pod did not become ready")
+		By("Visit metrics endpoint with TLS")
+		Eventually(func() error {
+			metricsOutput, _, err := Lexecute(ctx, clients.RestConfig, clients.KubeClient, operatorNamespace, podName, containerName,
+				[]string{
+					"/bin/sh", "-c",
+					fmt.Sprintf(
+						"curl -v -s --cacert %s/ca.crt -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" https://%s.%s.svc:8443/metrics",
+						certMountPath,
+						metricsServiceName,
+						operatorNamespace,
+					),
+				})
+			if err != nil {
+				return fmt.Errorf("exec into pod failed: %w", err)
+			}
+
+			if !strings.Contains(string(metricsOutput), "controller_runtime_reconcile_total") {
+				return fmt.Errorf("expected metric not found in output")
+			}
+			return nil
+		}, operatorReadyTime, operatorPoll).Should(Succeed(), "expected HTTP 200 OK from metrics endpoint")
+	})
 })
+
+func Lexecute(ctx context.Context, restConfig *rest.Config, kubeClient kubernetes.Interface, namespace, podName, containerName string, command []string) ([]byte, []byte, error) {
```

Comment:
Could we move the func to [testutils](https://github.com/openshift/lws-operator/tree/main/test/e2e/testutils)

Reply:
i think it's not very common function, so add here.